vignettes/secuTrialR-package-vignette.Rmd

```diff
@@ -1,172 +1,570 @@
 
-The secuTrial export has already been loaded above. The code below shows how to load the validation overview.
-The validation overview contains a line for every variable with a missing or erroneous value. As explained 
-earlier, patient 999 was added with missing values. As you can see the validation report only contains entries
-for "Patient" 999.
+## Conversion to SPSS, STATA, SAS
+
+Given that you are working with R it is unlikely that you need such conversions for
+yourself. However, collaborators may ask for data which is readily importable into SPSS,
+STATA or SAS. For this you
+can use `write_secuTrial()`.
+\vspace{5pt}
+
+```{r, eval = FALSE}
+# retrieve path to a temporary directory
+tdir <- tempdir()
+# write spss
+write_secuTrial(ctu05_data, format = "sav", path = tdir)
+```
+
+Since this has not been heavily tested or used there may be issues and
+you might prefer doing this manually with the `haven` package.
+
+\newpage
+
+## Subsetting `secuTrialdata`
+
+In some cases it may be useful to subset your `secuTrialdata` object.
+For example if you have cohort data and would like to supply a subset of
+the data for a retrospective study. We have implemented this
+option with `subset_secuTrial()`. It will truncate your `secuTrialdata` object
+and return a new `secuTrialdata` object which is a subset of the original data.
+It is possible to subset by including or excluding specific participant
+ids or centres.
+\vspace{5pt}
+
+```{r}
+# initialize some subset identifiers
+participants <- c("RPACK-INS-011", "RPACK-INS-014", "RPACK-INS-015")
+centres <- c("Inselspital Bern (RPACK)", "Universitätsspital Basel (RPACK)")
+
+# exclude Bern and Basel
+ctu05_data_berlin <- subset_secuTrial(ctu05_data, centre = centres, exclude = TRUE)
+# exclude Berlin
+ctu05_data_bern_basel <- subset_secuTrial(ctu05_data, centre = centres)
+# keep only subset of participants
+ctu05_data_pids <- subset_secuTrial(ctu05_data, participant = participants)
+
+class(ctu05_data_berlin)
+class(ctu05_data_bern_basel)
+class(ctu05_data_pids)
+```
+
+If you subset based on centres all traces of deleted centres will be removed.
+If you remove based on participant ids all traces of deleted participants will be removed.
+\vspace{5pt}
 
 ```{r}
-val_ovv <- read_validation_overview(data_dir = val_ovv_location)
-val_ovv[,c("Patient", "Completion status", "Column")]
+# only Berlin remains
+ctu05_data_berlin$ctr
+
+# all centres remain eventhough all three ids are from Bern
+ctu05_data_pids$ctr
+```
+
+\newpage
+
+Since the truncated object's class remains unchanged (i.e. `secuTrialdata`) you can
+still use the generic functions on it.
+Let's say you would only like to look at the recruitment plot for Bern alone.
+\vspace{5pt}
+
+```{r, fig.height = 3.8, fig.width = 8}
+# keep only Bern
+ctu05_data_bern <- subset_secuTrial(ctu05_data, centre = "Inselspital Bern (RPACK)")
+# plot
+plot_recruitment(ctu05_data_bern)
```

Comment:
Nothing wrong with the vignette here, just realized we might want to change the axis tick labels to actually include the date of enrolment. E.g. in this case not days of week, but instead dd-mm-yy (or dd-mm). What do you think? otherwise we've got axis label "Date of enrollment" e.g. "Thu". Also with "Thu", one has to go back to the export options and calculate what it refers to.

Reply:
Generally I agree and I have also "stumbled" over this, but the problem fixes itself once data has been captured over a period of several months/years (the norm). Thus, I would leave it as is.